tests/core/version/test_production_version_migration.py

```diff
@@ -264,7 +263,7 @@ def test_migrate_compatible_version():
         _ScenarioManager._submit(scenario_v2_1)
 
     assert scenario_v2_1.d2.read() == 6
-    assert len(_DataManager._get_all(version_number="all")) == 6
+    assert len(_DataManager._get_all(version_number="all")) == 10
```

Comment:
I'm not sure why would the number of data nodes increases here. I will investigate